client/components/stepper/test/stepper-panel.test.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import { StepperPanel } from '../stepper-panel';
+
+// Mock the Icon component to avoid dependency on WordPress icons
+jest.mock( 'wcpay/components/wp-components-wrapped', () => ( {
+	Icon: ( { size }: { size: number } ) => (
+		<span data-testid="mock-icon">icon-{ size }</span>
+	),
+} ) );
+
+describe( 'StepperPanel', () => {
+	it( 'renders all step labels', () => {
+		const steps = [ 'Step 1', 'Step 2', 'Step 3' ];
+		render( <StepperPanel steps={ steps } currentStep={ 1 } /> );
+		steps.forEach( ( label ) => {
+			expect( screen.getByText( label ) ).toBeInTheDocument();
+		} );
+	} );
+
+	it( 'marks previous steps as complete and current as active', () => {
+		const steps = [ 'A', 'B', 'C' ];
+		render( <StepperPanel steps={ steps } currentStep={ 1 } /> );
+		// Instead, check DOM classes
+		const allStepDivs = screen
+			.getAllByText( /A|B|C/ )
+			.map( ( labelNode ) => labelNode.closest( '.stepper-step' ) );
+		// Step 0 should be complete
+		expect( allStepDivs[ 0 ] ).toHaveClass( 'complete' );
+		// Step 1 should be active
+		expect( allStepDivs[ 1 ] ).toHaveClass( 'active' );
+		// Step 2 should be neither
+		expect( allStepDivs[ 2 ] ).not.toHaveClass( 'active' );
+		expect( allStepDivs[ 2 ] ).not.toHaveClass( 'complete' );
```

Comment:
( ⛏️ ) You could use array destructuring to be more explicit in the naming of the elements
```suggestion
		// Instead, check DOM classes
		const [ step1, step2, step3 ] = screen
			.getAllByText( /A|B|C/ )
			.map( ( labelNode ) => labelNode.closest( '.stepper-step' ) );
		// Step 0 should be complete
		expect( step1 ).toHaveClass( 'complete' );
		// Step 1 should be active
		expect( step2 ).toHaveClass( 'active' );
		// Step 2 should be neither
		expect( step3 ).not.toHaveClass( 'active' );
		expect( step3 ).not.toHaveClass( 'complete' );
```